Changes

vadrian89 opened the pull request:
High guys, 

Please check / test and let me know if you approve of this, or let me know issues to fix so I can merge it with Daryl's branch so he can make a new package.

Thanks!

Commit:
New weather API added and more

New:
- added API selector 
- api cannot be changed if a new location was searched
- temperature / speed is calculated when weather is downloaded based on
user selection and downloaded temperature / speed units   
- added function for temperature unit
- added ubuntu condensed font for international languages support  
- added weather description  
- added link in credits page for Open Weather Map provider 
- added weather description in system language   

Changed:
- ui is now flexible, the ui items aren't absolute positioned
anymore(this can help for future widget resizing option)    
- changed ubuntu condensed font with the newer version everywhere it was
used 
- changed behavior of sync button, the weather should be shown after the
sync has finished completely 
- increased weather update from 15 mins to 60 mins (because weather
doesn't usually changes so drastically in so little time span), weather
update on demand still works   
- fixed issue where default settings were not set on first run 
- settings and sync button arer hidden on first run 
- changed a bit font size and margins of detail divs to fit 
- moved javascript code in different source files and named accordingly
to be more readable

Known Issues:
- gradient background not setting correctly based on weather   
- location might bug out when toggling through API and / or changing
weather temperature / speed units(looking into it)  
- weather still fades in before weather is updated when loader button
has been clicked, on new location, sometimes 
- weather loader button(with tick) sometimes remains visible after
location has been changed

Commit:
Changed Readme file

Commit:
Changed readme to include proper link to Archisman Panigrahi's github
page

kd8bny commented:
Roger will do!

vadrian89 commented:
Apparently I was calculating the speed from Open Weather Map wrong because I was setting the unit as mph (don't know so dont ask me why :P) even though they send the default speed unit as m/s , changed it now and will commit this evening.

Edit: 
I am also thinking of hidding the buttons for "Support" and "How To" until they are implementing so they don't stir confusion amongst users.

kd8bny commented:
If you guys go ahead and build from a clean system (without cumulus install) do you recieve the following error?

"(cumulus:11030): GLib-GIO-ERROR **: Settings schema 'net.launchpad.cumulus' is not installed"

It isnt a huge deal you just have to recompile the glib schema. 
I think the issue is in the build file probably something I forgot to change awhile ago

ps the changes look great Adrian! Lets merge 'em!
I concur with hiding those 2 buttons too...Just till they do something

vadrian89 commented:
Yeah I think I had this error when trying to run on a clean system.

Btw I'm also looking into packaging/deploying for ubuntu but for today I'm
finished.
To much learning for today.

I'm at a point of making a new setup.py file / making a snap file / buildin
deb packages for ppa.
Something bugs me, what package should contain gi.repository?

Pe 25 sept. 2016 5:13 p.m., "Daryl Bennett" notifications@github.com a
scris:

> If you guys go ahead and build from a clean system (without cumulus
> install) do you recieve the following error?
> 
> "(cumulus:11030): GLib-GIO-ERROR **: Settings schema
> 'net.launchpad.cumulus' is not installed"
> 
> It isnt a huge deal you just have to recompile the glib schema.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/kd8bny/cumulus/pull/4#issuecomment-249423968, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/AHzIhiSE2dv71Oo_cdWUga8ZXnLm8qUuks5qtoFugaJpZM4KESIQ
> .

kd8bny commented:
What package of ours or what package gi.repository comes from?

vadrian89 commented:
What package is gi.repository coming from.

Here is my askubuntu question:
http://askubuntu.com/questions/829374/missing-dependency-in-python-application-packaged-with-snapcraft

2016-09-25 17:26 GMT+03:00 Daryl Bennett notifications@github.com:

> What package of ours or what package gi.repository comes from?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/kd8bny/cumulus/pull/4#issuecomment-249424643, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/AHzIhtxTn6MtcFgC5-GtjiSKUMbMUc9Zks5qtoR-gaJpZM4KESIQ
> .

kd8bny commented:
I think this is what youre looking for 

https://wiki.gnome.org/Projects/PyGObject#Binaries

apt-get install python-gi

vadrian89 commented:
Ill look a bit more later, now I'm relaxing :)

Pe 25 sept. 2016 5:29 p.m., "Daryl Bennett" notifications@github.com a
scris:

> I think this is what youre looking for
> 
> https://wiki.gnome.org/Projects/PyGObject#Binaries
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/kd8bny/cumulus/pull/4#issuecomment-249424836, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/AHzIhmdGnwhZuyONFI0Gxy3ocW9bwt7Hks5qtoVmgaJpZM4KESIQ
> .

vadrian89 commented:
Can i mail the setup.py so you guys have a look at it? And if needed it fix
it? :-)

Pe 25 sept. 2016 5:38 p.m., "Verban Adrian" verbanady@gmail.com a scris:

> Ill look a bit more later, now I'm relaxing :)
> 
> Pe 25 sept. 2016 5:29 p.m., "Daryl Bennett" notifications@github.com a
> scris:
> 
> > I think this is what youre looking for
> > 
> > https://wiki.gnome.org/Projects/PyGObject#Binaries
> > 
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > https://github.com/kd8bny/cumulus/pull/4#issuecomment-249424836, or mute
> > the thread
> > https://github.com/notifications/unsubscribe-auth/AHzIhmdGnwhZuyONFI0Gxy3ocW9bwt7Hks5qtoVmgaJpZM4KESIQ
> > .

kd8bny commented:
Of course haha...enjoy your  relax time.
Its morning here so I have all day lol

vadrian89 commented:
For me it 5:42 PM. :), I'll send it in 10 mins

Pe 25 sept. 2016 5:41 p.m., "Daryl Bennett" notifications@github.com a
scris:

> Of course haha...enjoy your relax time.
> Its morning here so I have all day lol
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/kd8bny/cumulus/pull/4#issuecomment-249425477, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/AHzIhvf1wfiMIglLhI5YSCGIZNL1dyq3ks5qtog2gaJpZM4KESIQ
> .

vadrian89 commented:
Here is the setup.py I came up atm.

2016-09-25 17:43 GMT+03:00 Verban Adrian verbanady@gmail.com:

> For me it 5:42 PM. :), I'll send it in 10 mins
> 
> Pe 25 sept. 2016 5:41 p.m., "Daryl Bennett" notifications@github.com a
> scris:
> 
> > Of course haha...enjoy your relax time.
> > Its morning here so I have all day lol
> > 
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > https://github.com/kd8bny/cumulus/pull/4#issuecomment-249425477, or mute
> > the thread
> > https://github.com/notifications/unsubscribe-auth/AHzIhvf1wfiMIglLhI5YSCGIZNL1dyq3ks5qtog2gaJpZM4KESIQ
> > .

Commit:
Changes & fixes  

Changes: 
- changed schema to match cumulus project
- changed setup.py completely with a standard script

Fixes: 
- fixed issue where the script for open weather map calculated the
weather speed wrong  
- changed the way weather updates inteval because the interval was never
resetting, just creating more update intervals in case operations such
as (changed location / api / measurements), now it should properly
update the weather on an hour basis since last update  

Added: 
- cumulus.desktop file
- snapcraft.yaml file

vadrian89 commented:
Made some progress yesterday evening, though I had little time (about 2 hours), I will continue this evening maybe I will manage to pull it through.  
Need to remake the tarball structure made from the setup.py, for a ppa I need to build it Ubuntu tools so debian documents need to be avoided. 

Anyway, the general ideea is that I'm making progress and if I don't manage to make a debian package for the ppa by the end of this day I will surely finish by the end of this week.
After I manage to succede with debian I will continue with snapcraft.

vadrian89 commented on data/glib-2.0/schemas/com.github.cumulus.gschema.xml:
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <schemalist gettext-domain="cumulus">
-  <schema id="net.launchpad.cumulus" path="/net/launchpad/cumulus/">
+  <schema id="com.github.cumulus" path="/apps/cumulus/" gettext-domain="cumulus">
```
need to remove one gettext-domain="cumulus" as it already was in schemalist, don't know how I missed it.

archisman-panigrahi commented:
Hi, I am really sorry to have kept you waiting, but, I have ported Cumulus to Quickly. I need a few more days to set up License etc. before I can upload to the PPA but it is quite ready. I am using master branch of @vadrian89, and found a peculiar bug. Some opacity values are not as expected. They are 
o0.1
o0.15000000000000002
o0.2
o0.25
o0.30000000000000004
o0.35
o0.4
o0.45000000000000007
o0.5
o0.55
o0.6
o0.65
o0.7000000000000001
o0.75
o0.8
o0.85
o0.9
o0.9500000000000001
o1
instead of expected o0.15 etc. And, opacity of those values are not working as expected.
I tried to round values as described [here](http://stackoverflow.com/a/11832950/3128341), but that does not work.

![screenshot from 2016-09-28 19 34 08](https://cloud.githubusercontent.com/assets/3063132/18917181/9b6c0418-85b4-11e6-8831-58e8182dae7f.png)

vadrian89 commented:
It's ok, just came home from work will look into the issue rigtht away.

Pe 28 sept. 2016 5:29 p.m., "Archisman Panigrahi" notifications@github.com
a scris:

> Hi, I am really sorry to have kept you waiting, but, I have ported Cumulus
> to Quickly. I need a few more days to set up License etc. before I can
> upload to the PPA but it is quite ready. I am using master branch of
> @vadrian89 https://github.com/vadrian89, and found a peculiar bug. Some
> opacity values are not as expected. They are
> o0.1
> o0.15000000000000002
> o0.2
> o0.25
> o0.30000000000000004
> o0.35
> o0.4
> o0.45000000000000007
> o0.5
> o0.55
> o0.6
> o0.65
> o0.7000000000000001
> o0.75
> o0.8
> o0.85
> o0.9
> o0.9500000000000001
> o1
> instead of expected o0.15 etc. And, opacity of those values are not
> working as expected.
> I tried to round values as described here
> http://stackoverflow.com/a/11832950/3128341, but that does not work.
> 
> [image: screenshot from 2016-09-28 19 34 08]
> https://cloud.githubusercontent.com/assets/3063132/18917181/9b6c0418-85b4-11e6-8831-58e8182dae7f.png
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/kd8bny/cumulus/pull/4#issuecomment-250183044, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/AHzIhrws_kx6cd3SCvvtw_wKSLXd5udGks5qunmsgaJpZM4KESIQ
> .

archisman-panigrahi commented:
Check this  https://launchpad.net/~cumulus-team/+archive/ubuntu/cumulus

archisman-panigrahi commented:
@vadrian89 I just used Quickly template, changed necessary files and copied files from `data/media` of your master branch. System language has to be manually added to Python. Also, I have not tried your setup.py, and used the setup.py created by Quickly. We will hve to try to have one source code that will work with both Debian and Snap packaging system, or otherwise will have to maintain two branches.

PreferencesDialog.py was added by Quickly and probably, can safely be removed

archisman-panigrahi commented:
I have updated setup.py (will push soon). @vadrian89 How to round opacity values upto 2 decimal places? I tried

```
function updateTitle(val) {
    document.title = "o" + Math.Round(val*100)/100;
    localStorage.app_opacity = Math.Round(val*100)/100;
}
```

But that just stops updating Title when opacity is change

vadrian89 commented:
Can you please mail me the setup.py file from Quickly?

Pe 29 sept. 2016 9:43 a.m., "Archisman Panigrahi" notifications@github.com
a scris:

> @vadrian89 https://github.com/vadrian89 I just used Quickly template,
> changed necessary files and copied files from data/media of your master
> branch. System language has to be manually added to Python. Also, I have
> not tried your setup.py, and used the setup.py created by Quickly. We will
> hve to try to have one source code that will work with both Debian and Snap
> packaging system, or otherwise will have to maintain two branches
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/kd8bny/cumulus/pull/4#issuecomment-250383695, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/AHzIhq5hpJuq1nKBGiK35c0tZcut3pdMks5qu14BgaJpZM4KESIQ
> .

archisman-panigrahi commented:
I have fixed it. Will push the commit when this Pull request is accepted. I was doing `Math.Round` instead of `Math.round`, which created the problem

vadrian89 commented:
Ok, I will accept the pull request this evening after taking another look
into it, now I'm at work and a bit busy.

Pe 29 sept. 2016 10:35 a.m., "Archisman Panigrahi" notifications@github.com
a scris:

> I have fixed it. Will push the commit when this Pull request is accepted.
> I was doing Math.Round instead of Math.round, which created the problem
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/kd8bny/cumulus/pull/4#issuecomment-250392743, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/AHzIhgn0gRZ6C_QfxxO_ca2th1L8efmUks5qu2okgaJpZM4KESIQ
> .

Commit:
Changed schema and changelog
